smithy-model/src/main/java/software/amazon/smithy/model/loader/LoadOperationProcessor.java

```diff
@@ -169,7 +169,7 @@ private void detectAndEmitForwardReference(LoadOperation.ForwardReference refere
         } else {
             // Try to find a prelude shape by ID if no ID exists in the namespace with this name.
             ShapeId preludeId = ShapeId.fromOptionalNamespace(Prelude.NAMESPACE, reference.name);
-            if (prelude.getShapeIds().contains(preludeId)) {
+            if (prelude != null && prelude.getShapeIds().contains(preludeId)) {
```

Comment:
Is there a case where prelude will not be available?

Reply:
Good catch. Added.